sdk/eventhub/azure-eventhub-checkpointstoretable/tests/test_storage_table_partition_manager.py

```diff
@@ -1,8 +1,148 @@
 import pytest
 from azure.eventhub.extensions.checkpointstoretable import TableCheckpointStore
+from azure.data.tables import TableServiceClient
+import uuid
+import warnings
+import random
+import os
+from azure.eventhub.exceptions import OwnershipLostError
 
+STORAGE_CONN_STR = [os.environ.get("AZURE_STORAGE_CONN_STR", "Azure Storage Connection String")]
 
-def test_constructor():
-    client = TableCheckpointStore()
-    assert client is not None
-    
+def get_live_storage_table_client(storage_connection_str):
+    try:
+        table_name = chr(random.randint(ord('a'), ord('z'))) +(uuid.uuid4().hex)
+        table_service_client = TableServiceClient.from_connection_string(storage_connection_str)
+        table_service_client.create_table_if_not_exists(table_name)
+        return storage_connection_str, table_name
+    except:
+        pytest.skip("Storage table client can't be created")
+
+def remove_live_storage_table_client(storage_connection_str, table_name):
+    try:
+        table_service_client = TableServiceClient.from_connection_string(storage_connection_str)
+        table_service_client.delete_table(table_name)
+    except:
+        warnings.warn(UserWarning("storage table teardown failed"))
+
+def _claim_and_list_ownership(storage_connection_str, table_name):
+    fully_qualified_namespace = 'test_namespace'
+    eventhub_name = 'eventhub'
+    consumer_group = '$default'
+    ownership_cnt = 8
+
+    checkpoint_store = TableCheckpointStore.from_connection_string(
+        storage_connection_str, table_name)
+    ownership_list = checkpoint_store.list_ownership(
+            fully_qualified_namespace,
+            eventhub_name,
+            consumer_group)
+    assert len(ownership_list) == 0
+
+    ownership_list = []
+
+    for i in range(ownership_cnt):
+            ownership = {}
+            ownership['fully_qualified_namespace'] = fully_qualified_namespace
+            ownership['eventhub_name'] = eventhub_name
+            ownership['consumer_group'] = consumer_group
+            ownership['owner_id'] = 'owner_id'
+            ownership['partition_id'] = str(i)
+            ownership['etag'] = None
+            ownership['last_modified_time'] = None
+            ownership_list.append(ownership)
+    newownershiplist = checkpoint_store.claim_ownership(ownership_list)
+    assert len(newownershiplist) == len(ownership_list)
+    for i in range(len(ownership_list)):
+        assert ownership_list[i]['etag'] != newownershiplist[i]['etag']
+        assert ownership_list[i]['last_modified_time'] != newownershiplist[i]['last_modified_time']
+
+    ownership_list = checkpoint_store.list_ownership(fully_qualified_namespace,eventhub_name,consumer_group)
+    assert len(ownership_list) == ownership_cnt
+    assert len(ownership_list) == len(newownershiplist)
+    for i in range(len(newownershiplist)):
+        assert ownership_list[i]['last_modified_time'] == newownershiplist[i]['last_modified_time']
```

Comment:
I think this is also where you'd want to check the etag for each ownership is unchanged.

Reply:
oh yes I forgot

Reply:
on your comment about the offset it should be changed to int